### PR TITLE
SCP -1289 untyped vs typed CEK in plutus-core-test NEAT

### PR DESCRIPTION
**Pull request by jmchapman**
This PR simply adds a check that

`erase . typed_CEK == untyped_CEK . erase` 

in the NEAT tests of `plutus-core-test`.

Pre-submit checklist:
- Branch
    - [x] Commit sequence broadly makes sense
    - [x] Key commits have useful messages
    - [x] Relevant tickets are mentioned in commit messages
- PR
    - [x] Self-reviewed the diff
    - [x] Useful pull request description
    - [x] Reviewer requested
- If you updated any cabal files or added Haskell packages:
    - [x] `nix-shell shell.nix --run updateMaterialized` to update the materialized Nix files
    - [x] Update `hie-*.yaml` files if needed
- If you changed any Haskell files:
    - [x] `nix-shell shell.nix --run fix-stylish-haskell` to fix any formatting issues
- If you changed any Purescript files:
    - [x] `nix-shell shell.nix --run fix-purty` to fix any formatting issues

Pre-merge checklist:
- [x] Someone approved it
- [x] Commits have useful messages
- [x] Review clarifications made it into the code
- [x] History is moderately tidy; or going to squash-merge

### CODE REVIEW

**Comment by jmchapman** [comment]
77145fb timed out on the cost model. I wonder if the a4bca4c might have more luck.

File: plutus-core/generators/Language/PlutusCore/Generators/NEAT/Spec.hs [hunk 1]
```diff
@@ -166,13 +177,27 @@ prop_agree_Ck_Cek tyG tmG = do
...
   tmCek <- withExceptT CekP $ liftEither $
     evaluateCek testBuiltinsRuntime tm `catchError` handleError ty
 
   tmCk <- withExceptT CkP $ liftEither $
     evaluateCk testBuiltinsRuntime tm `catchError` handleError ty
-  unless (tmCk == tmCek) $ throwCtrex (CtrexTermEvaluationMismatch tyG tmG [tmCek,tmCk])
+
+  -- check if CK and CEK give the same output
+  unless (tmCk == tmCek) $
+    throwCtrex (CtrexTermEvaluationMismatch tyG tmG [tmCek,tmCk])
```
Comment (effectfully) [inline plutus-core/generators/Language/PlutusCore/Generators/NEAT/Spec.hs:188]: Do they? The typed CEK machine does not handle types correctly while the CK machine does. Please remind me how this discrepancy is resolved in NEAT tests.

File: plutus-core/generators/Language/PlutusCore/Generators/NEAT/Spec.hs [hunk 2]
```diff
@@ -166,13 +177,27 @@ prop_agree_Ck_Cek tyG tmG = do
...
+
+  -- erase CEK output
+  let tmCekU = U.erase tmCek
+
+  -- run untyped CEK on erased input
+  tmUCek <- withExceptT UCekP $ liftEither $
+    U.evaluateCek testBuiltinsRuntime (U.erase tm) `catchError` handleUError
```
Comment (effectfully) [inline plutus-core/generators/Language/PlutusCore/Generators/NEAT/Spec.hs:195]: ``blah `catchError` handleUError`` is called for all the machines. Could we drop it then and still expect the results to be equal?

**Review by effectfully — APPROVED** [review]

File: plutus-core/generators/Language/PlutusCore/Generators/NEAT/Spec.hs (hunk 1 above)
Comment (jmchapman) [reply plutus-core/generators/Language/PlutusCore/Generators/NEAT/Spec.hs:188]: It only produces something at type unit which has no type annotations.

File: plutus-core/generators/Language/PlutusCore/Generators/NEAT/Spec.hs (hunk 2 above)
Comment (jmchapman) [reply plutus-core/generators/Language/PlutusCore/Generators/NEAT/Spec.hs:195]: I don't want a user error to be considered a counter example/test failure found be NEAT. This handler catches user errors and turns them the error terms which are returned 'successfully'.

Comment (effectfully) [reply plutus-core/generators/Language/PlutusCore/Generators/NEAT/Spec.hs:195]: Ah, sure, silly me.

File: plutus-core/generators/Language/PlutusCore/Generators/NEAT/Spec.hs (hunk 1 above)
Comment (effectfully) [reply plutus-core/generators/Language/PlutusCore/Generators/NEAT/Spec.hs:188]: I see. Is that documented?

Comment (jmchapman) [reply plutus-core/generators/Language/PlutusCore/Generators/NEAT/Spec.hs:189]: Good point. I have added some comments and removed an misleading one.

**Comment by jmchapman** [comment]
I tried to stop it upsetting haddock and now it seems to be timing out on the cost model again.  Trying again with a gratuitous `git commit --amend`.

**Pull request merged**